File: Python/1048 - Salary Increase.py
salary = float(input())
adj = [15,12,10,7,4]
if salary <= 400:
    final_salary = salary * (1+(adj[0]/100))
    inc = final_salary - salary
    print("Novo salario: %.2f" % final_salary)
    print("Reajuste ganho: %.2f" % inc)
    print("Em percentual: %d %%" % adj[0])
elif salary > 400 and salary <= 800:
    final_salary = salary * (1+(adj[1]/100))
    inc = final_salary - salary
    print("Novo salario: %.2f" % final_salary)
    print("Reajuste ganho: %.2f" % inc)
    print("Em percentual: %d %%" % adj[1])
elif salary > 800 and salary <= 1200:
    final_salary = salary * (1+(adj[2]/100))
    inc = final_salary - salary
    print("Novo salario: %.2f" % final_salary)
    print("Reajuste ganho: %.2f" % inc)
    print("Em percentual: %d %%" % adj[2])
elif salary > 1200 and salary <= 2000:
    final_salary = salary * (1+(adj[3]/100))
    inc = final_salary - salary
    print("Novo salario: %.2f" % final_salary)
    print("Reajuste ganho: %.2f" % inc)
    print("Em percentual: %d %%" % adj[3])
else:
    final_salary = salary * (1+(adj[4]/100))
    inc = final_salary - salary
    print("Novo salario: %.2f" % final_salary)
    print("Reajuste ganho: %.2f" % inc)
    print("Em percentual: %d %%" % adj[4])

File: C++/1048 - Salary Increase.cpp
#include <iostream>
#include <vector>

int get_wage_group(double wage)
{
    if (wage >= 0 && wage <= 400)
    {
        return 0;
    }
    if (wage > 400 && wage <= 800)
    {
        return 1;
    }
    if (wage > 800 && wage <= 1200)
    {
        return 2;
    }
    if(wage > 1200 && wage <= 2000)
    {
        return 3;
    }
    return 4;
}

void get_raise(double wage, int group, std::vector<int> raise_rate)
{
    double new_wage = wage*((100+raise_rate[group])/100.00);
    double raise_value = new_wage-wage;
    printf("Novo salario: %.2f\n",new_wage);
    printf("Reajuste ganho: %.2f\n",raise_value);
    printf("Em percentual: %d %%\n",raise_rate[group]);
}

int main()
{
    std::vector<int> raise_rate = {15, 12, 10, 7, 4};
    double wage;
    std::cin >> wage;
    int group = get_wage_group(wage);
    get_raise(wage,group,raise_rate);
    return 0;
}